### PR TITLE
add typescript docs

### PR DESCRIPTION
**Pull request by tsiq-swyx**
Issue: https://github.com/storybooks/storybook/issues/3270

## What I did

Added typescript docs

## How to test

its just docs :)

### CODE REVIEW

**Comment by tsiq-swyx** [comment]
cc @danielduan

File: docs/src/pages/configurations/custom-webpack-config/index.md [hunk 1]
```diff
@@ -96,28 +96,30 @@ If so, this is how you do it using the Full Control Mode.
...
+
 ## Using Your Existing Config
 
 You may have an existing Webpack config for your project. So, you may need to copy and paste some config items into Storybook's custom Webpack config file.
 
 But you don't need to. There are a few options:
 
--   Simply import your main Webpack config into Storybook's `webpack.config.js` and use the loaders and plugins used in that.
--   Create a new file with common Webpack options and use it in both inside the main Webpack config and inside Storybook's `webpack.config.js`.
+* Simply import your main Webpack config into Storybook's `webpack.config.js` and use the loaders and plugins used in that.
```
Comment (danielduan) [inline docs/src/pages/configurations/custom-webpack-config/index.md]: Can you remove this `Simply?`. If it's so simple, users wouldn't need to read the docs.

It should just say `Import your main Webpack...`

File: docs/src/pages/configurations/custom-webpack-config/index.md [hunk 2]
```diff
@@ -19,19 +19,19 @@ Let's say you want to add [SASS](http://sass-lang.com/) support to Storybook. Th
 Simply add the following content to a file called `webpack.config.js` in your Storybook config directory (`.storybook` by default ).
```
Comment (danielduan) [inline docs/src/pages/configurations/custom-webpack-config/index.md]: same with this `Simply`

File: docs/src/pages/configurations/typescript-config/index.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+title: 'Typescript Config'
+---
+
+This is a central reference for using Storybook with Typescript.
+
+## Dependencies you may need
+
+```bash
+yarn add -D typescript
+yarn add -D awesome-typescript-loader # alternative to ts-loader
```
Comment (danielduan) [inline docs/src/pages/configurations/typescript-config/index.md]: are these two loaders interchangeable in this guide? we should leave a note about it.

File: docs/src/pages/configurations/typescript-config/index.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+yarn add -D @storybook/addon-info react-docgen-typescript-webpack-plugin # optional but recommended
+yarn add -D jest "@types/jest" ts-jest #testing
+```
+
+## Setting up Typescript to work with Storybook
+
+We first have to use the [custom Webpack config in full control mode, extending default configs](https://storybook.js.org/configurations/custom-webpack-config/#full-control-mode--default):
```
Comment (danielduan) [inline docs/src/pages/configurations/typescript-config/index.md]: I think relative URLs work here, so `/configurations/custom...`

File: docs/src/pages/configurations/typescript-config/index.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    "copy-woff2-to-lib": "cpx \"./src/**/*.woff2\" ./build/lib",
+    "copy-svg-to-lib": "cpx \"./src/**/*.svg\" ./build/lib",
+    "copy-png-to-lib": "cpx \"./src/**/*.png\" ./build/lib",
+    "lint": "tslint -c tslint.json 'src/**/*.{ts,tsx}'"
+  },
+```
+
+## Related Issues and Helpful Resources
+
+* [Issue: Add proper typescript webpack setup in readme #1778](https://github.com/storybooks/storybook/issues/1778)
```
Comment (danielduan) [inline docs/src/pages/configurations/typescript-config/index.md]: Let's leave these GitHub issues out of the docs. If you think there are useful information there, please copy and paste them to the doc instead.

**Review by danielduan — CHANGES_REQUESTED** [review]
Thanks for the PR, really appreciate it! Just some minor fixes and we're good to go.

File: docs/src/pages/configurations/custom-webpack-config/index.md (hunk 1 above)
Comment (tsiq-swyx) [reply docs/src/pages/configurations/custom-webpack-config/index.md]: hahaha i didnt write that

File: docs/src/pages/configurations/custom-webpack-config/index.md (hunk 2 above)
Comment (tsiq-swyx) [reply docs/src/pages/configurations/custom-webpack-config/index.md]: hahaha i didnt write that

File: docs/src/pages/configurations/typescript-config/index.md (hunk 3 above)
Comment (tsiq-swyx) [reply docs/src/pages/configurations/typescript-config/index.md]: yes

File: docs/src/pages/configurations/typescript-config/index.md (hunk 4 above)
Comment (tsiq-swyx) [reply docs/src/pages/configurations/typescript-config/index.md]: ok

**Comment by tsiq-swyx** [comment]
I have updated with your requested fixes

**Comment by estaub** [comment]
The thing I really liked about https://medium.com/@mtiller/storybook-react-typescript-and-jest-c9059ea06fa7 was that it provided a clear (but slightly defective) set of instructions that I could quickly copy/paste from and be running.  You've deliberately avoided this with the webpack config, thinking folks should use what they use elsewhere.  But there's a lot in a good production webpack config that's not relevant at all, and may break in this context; think about providing a bare-bones example `.storybook/webpack.config.js for typescript-loader and/or a-t-l.  Example:

```js
const genDefaultConfig = require('@storybook/react/dist/server/config/defaults/webpack.config.js');
module.exports = (baseConfig, env) => {
    const config = genDefaultConfig(baseConfig, env);
    // Add typescript loader:
    config.module.rules.push({
        test: /\.(ts|tsx)$/,
        loader: require.resolve('awesome-typescript-loader')
    });
    config.resolve.extensions.push('.ts', '.tsx');
    return config;
}
```

**Comment by estaub** [comment]
In the tsconfig.json section, you've buried the lede: the main copy/paste content should work for the default configuration, where `stories` is a peer of `src`.  The main takeaway there (I think) is to be sure to include the stories; this shouldn't be only in an easily-overlooked sentence below the code.

**Comment by tsiq-swyx** [comment]
I also wrote this https://dev.to/swyx/quick-guide-to-setup-your-react--typescript-storybook-design-system-1c51

**Review by danielduan — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
@tsiq-swyx This is absolutely epic! Thank you a ton 🙇 

When my batch of stickers arrive, I'm sending you some, contact me via twitter or slack to send me your address, if you'd like them 👍

**Comment by tsiq-swyx** [comment]
oh haha cant say no to that. twitter it is

**Comment by sw-yx** [comment]
(not at all urgent, just curious) Whats the process for this being deployed to the live docs site? you guys dont use netlify for CD?

**Comment by Hypnosphi** [comment]
We deploy from release branch to avoid confusion, so I'll just pick this pr there now, and it'll be deployed authomatically

**Comment by Hypnosphi** [comment]
And it's live https://storybook.js.org/configurations/typescript-config/

**Pull request merged**